addons/binding/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/HttpChannelState.java

```diff
@@ -0,0 +1,246 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.http.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.HSBType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.library.types.PointType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.types.UpDownType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.core.transform.TransformationException;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.TypeParser;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.http.internal.model.CommandRequest;
+import org.openhab.binding.http.internal.model.ErrorListener;
+import org.openhab.binding.http.internal.model.StateRequest;
+import org.openhab.binding.http.internal.model.Transform;
+import org.openhab.binding.http.internal.util.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+
+import static org.openhab.binding.http.internal.HttpBindingConstants.CHANNEL_TYPE_ID_IMAGE;
+import static org.openhab.binding.http.internal.HttpBindingConstants.DEFAULT_CONTENT_TYPE;
+
+/**
+ * This holds the state and handles commands for a channel connected to a HTTP Thing.
+ *
+ * @author Brian J. Tarricone - Initial contribution
+ */
+@NonNullByDefault
+public class HttpChannelState implements AutoCloseable {
+    private static final List<Class<? extends State>> STATE_TYPES = Arrays.asList(
+            OnOffType.class,
+            OpenClosedType.class,
+            UpDownType.class,
+            PointType.class,
+            HSBType.class,
+            DecimalType.class,
+            DateTimeType.class,
+            StringType.class
+    );
```

Comment:
This is the last actual error that I spotted. You are using a fixed list of STATE_TYPES.
But depending on your channel type you need slightly different lists.

For example the PointType has a higher priority then the HSBType in your list. But the point type will match any received string of the format "x,y,z", so the HSBType will never match and your color channel will end up with a PointType.

I'm afraid you need to define multiple lists (best do that in the Constants class), and depending on the channel type (channelTypeUID) you use the fitting list